infra/scripts/replace-running-components.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+# Copyright 2018 VMware, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+for x in $(echo GOVC_USERNAME GOVC_PASSWORD VCH_NAME GOVC_URL GOPATH GOVC_INSECURE); do
+    if [[ ! -v $x ]]; then
+        echo "Insufficient inputs. Please set $x environment variable and re-execute this script.";
+        echo "GOVC_USERNAME: username on ESX/vCenter target"
+        echo "GOVC_PASSWORD: password on ESX/vCenter target"
+        echo "VCH_NAME: name of VCH; matches --name argument for vic-machine"
+        echo "GOVC_URL: IP or FQDN of your vCenter/ESX target"
+        echo "GOPATH: your GOPATH, obviously"
+        echo "GOVC_INSECURE: set to 1 to disable tls verify when using govc to talk to ESX/vC"
+        exit 1
+    fi;
+done
+
+function get-thumbprint () {
+    openssl s_client -connect $GOVC_URL:443 </dev/null 2>/dev/null \
```

Comment:
check out `govc about.cert --help`, particularly `govc about.cert --thumbprint`
It removes the additional dependency on having openssl available (I know it's very common but I'd prefer to assume `govc`.